sys/build.rs

```diff
@@ -1,31 +1,239 @@
-extern crate pkg_config;
+use semver::Version;
```

Comment:
I think you're trying to do something way too clever. Apart from the static build, we shouldn't do anything in here. Since we're linking dynamically, it should be up to the users to decide what they want to do. And it's even worse in case of cross-compilation...

Reply:
Sorry, I forgot we still needed to link. So apart from the `pkg-config` (and the equivalent for older versions) part, it shouldn't try to enforce the lib version with what the users have installed.

Reply:
That seems highly strange. You can only add new API without a major release, not delete or update the existing ones. Do you have examples where it's the case?

Reply:
I think this is expected behavior: the bindings have functions that may not be available in an older version of GEOS, so if you use a newer binding with older GEOS, you get linking errors.

For example, I installed GEOS 3.6.5 locally and attempted to test `cargo test --features v3_7_0`, which failed some tests with linkage errors:

```
Undefined symbols for architecture x86_64:
            "_GEOSReverse_r", referenced from:
                geos::geometry::Geometry::reverse::he885f71ff5967047 in libgeos-c9d2f998aeb36308.rlib(geos-c9d2f998aeb36308.4rsga35n89u0uz61.rcgu.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

In this case `GEOSReverse` is a function added in GEOS 3.7 and not available in my local 3.6 GEOS.

Reply:
This case should be covered by `pkg`, no?

Reply:
We can use `pkg-config` to enforce a minimum version; right now I'm setting that to the minimum supported in general (3.6.0).

However, we still need to verify version if set via environment variable or if the script falls back to `geos-config`, so it seemed like the cleanest way to handle that was to a have a single place where we compare the version they request via feature with the version detected by `pkg-config` / `geos-config` or provided via `GEOS_VERSION` env.  It also allows us to have a somewhat more helpful message than you might get if we rely on `pkg-config` to enforce the minimum version.

Reply:
If you specify `v3_7`, then you must have a libgeos 3.7 at minimum. But maybe it's what you're already doing. If so, sorry the time loss...

Reply:
This is indeed what we are doing [here](https://github.com/georust/geos/pull/113/files#diff-d8a3ade72837649313423dc94ecc4ee5cfa1872609bab33a8d4fc70f8b041589R201-R202); we're just not using `pkg-config` to enforce that - we're doing it in a single place for `pkg-config`, `geos-config`, and `GEOS_VERSION` env.